etl/steps/data/garden/war/2024-01-11/spread_of_nuclear_weapons.py

```diff
@@ -0,0 +1,133 @@
+"""Load a meadow dataset and create a garden dataset."""
+
+from owid.catalog import Table
+
+from etl.data_helpers import geo
+from etl.helpers import PathFinder, create_dataset
+
+# Get paths and naming conventions for current step.
+paths = PathFinder(__file__)
+
+# Latest year to be assumed for the content of the data, when intervals are open, e.g. "2000-", or "1980-".
+LATEST_YEAR = 2017
+
+
+def extract_year_ranges(years_ranges):
+    # Extract years from a string that contains a list of years and year ranges.
+    # Examples: "1964-66,72-75,80-", "1980-", "1953-62,82-87", "1970-2003", "1975-90", "2002-07".
+    years = []
+    if len(years_ranges) > 0:
+        for years_range in years_ranges.split(","):
+            if "-" in years_range:
+                start, end = years_range.split("-")
+                if len(end) == 0:
+                    end = str(LATEST_YEAR)
+                if len(start) == 2:
+                    start = "19" + start
+                if len(end) == 2:
+                    end = start[0:2] + end
+                years.extend(list(range(int(start), int(end) + 1)))
+            else:
+                years.append(int(years_range))
+
+    return years
+
+
+def correct_historical_regions(tb: Table) -> Table:
+    # Convert columns to string type, to avoid issues with categorical columns.
+    tb = tb.astype(str)
+
+    # Merge the years of nuclear weapons exploration of Germany and West Germany.
+    tb.loc[tb["country"] == "Germany", "explore"] = (
+        tb[tb["country"] == "Germany"]["explore"].astype(str).item()
+        + ","
+        + tb[tb["country"] == "West Germany"]["explore"].astype(str).item()
+    )
+    # Remove row for West Germany.
+    tb = tb[tb["country"] != "West Germany"].reset_index(drop=True)
+
+    # Assign the years of nuclear weapons exploration and pursuit of Yugoslavia to Serbia.
+    # To do that, simply rename the country.
+    tb.loc[tb["country"] == "Yugoslavia", "country"] = "Serbia"
```

Comment:
I agree this one is not very clear. I did this, firstly, because that's how it was done in the current version in grapher (so I assume Bastian put some thought on that already), and secondly, because it would be inconsistent to keep Yugoslavia when there are no other historical regions (e.g. there is no USSR). But Bastian and I are currently discussing whether we should include historical regions.